### PR TITLE
[MBL-17897][Student] Smart Search bar

### PR DESCRIPTION
**Pull request by hermannakos**
Test plan: Check the smart search bar on the course details page. Don't forget to enable the feature flag and allow the tab in the course. This ticket is for the search bar only, no search happens, no result screen, just a simple search bar.

refs: MBL-17897
affects: Student
release note: none

## Screenshots

<table>
<tr><th>Student</th>
<tr>
<td></td>
</tr>
<tr>
<td></td>
</tr>
</table>

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] Tested in dark mode
- [x] Tested in light mode
- [x] A11y checked
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

File: libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/SearchBar.kt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                keyboardOptions = KeyboardOptions.Default.copy(
+                    imeAction = ImeAction.Search
+                ),
+                keyboardActions = KeyboardActions(
+                    onSearch = {
+                        keyboardController?.hide()
+                        onSearch(query)
+                    }
+                ),
+                textStyle = TextStyle(
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/compose/composables/SearchBar.kt:100]: This might be an issue. See MBL-18086.

**Review by tamaskozmer — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Pull request merged**